benchmarks/src/main/java/zipkin2/server/internal/throttle/ThrottledCallBenchmarks.java

```diff
@@ -0,0 +1,112 @@
+ * Copyright 2015-2019 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin2.server.internal.throttle;
+
+import com.linecorp.armeria.common.metric.NoopMeterRegistry;
+import com.netflix.concurrency.limits.limit.FixedLimit;
+import com.netflix.concurrency.limits.limiter.SimpleLimiter;
+import java.io.IOException;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+import zipkin2.Call;
+import zipkin2.Callback;
+
+@Measurement(iterations = 5, time = 1)
+@Warmup(iterations = 10, time = 1)
+@Fork(3)
+@BenchmarkMode(Mode.SampleTime)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@State(Scope.Thread)
+@Threads(2)
+public class ThrottledCallBenchmarks {
+  ExecutorService fakeCallExecutor = Executors.newSingleThreadExecutor();
+  ExecutorService executor = Executors.newSingleThreadExecutor();
+  ThrottledCall call;
+
+  @Setup public void setup() {
+    executor = Executors.newSingleThreadExecutor();
+    fakeCallExecutor = Executors.newSingleThreadExecutor();
+    SimpleLimiter<Void> limiter = SimpleLimiter.newBuilder().limit(FixedLimit.of(1)).build();
+    LimiterMetrics metrics = new LimiterMetrics(NoopMeterRegistry.get());
+    Predicate<Throwable> isOverCapacity = RejectedExecutionException.class::isInstance;
+    call =
+      new ThrottledCall(new FakeCall(fakeCallExecutor), executor, limiter, metrics, isOverCapacity);
+  }
+
+  @TearDown public void tearDown() {
+    executor.shutdown();
+    fakeCallExecutor.shutdown();
+  }
+
+  @Benchmark public Void execute() throws IOException {
```

Comment:
In order to avoid oddities that occur between JVM boots you should add a second \@Benchmark which would test the second implementation under the same circumstances.  Basically, do a ThrottledCall and ThrottledCall2 impl.

Also, and I never did figure out how to do to this properly so please let me know if you figure it out, but you should test this with the G1GC compiler as well since that is the default in more modern JVMs.  I think I succeeded at doing that once by simply switching to the JRE11.